.github/workflows/check.yml

```
@@ -32,44 +31,61 @@ jobs:
         - 3.8
         - 3.7
         - 3.6
+        - 3.5
+        - pypy3
         - 2.7
         - pypy2
-        - pypy3
-
     steps:
-    - uses: actions/checkout@v2
-    - uses: tox-dev/action@v1.0.0
-      with:
-        python-version: ${{ matrix.py }}
-
+    - name: install OS dependencies
+      run:   |
+        if [ "$RUNNER_OS" == "Linux" ]; then
+             sudo apt-get update -y && sudo apt-get install fish csh
+        elif [ "$RUNNER_OS" == "Windows" ]; then
+             choco install vcpython27 --yes -f
+        elif [ "$RUNNER_OS" == "macOS" ]; then
+             brew update -vvv && brew install fish tcsh
+        fi
+      shell: bash
     - name: setup python for tox
       uses: actions/setup-python@v2
       with:
         python-version: 3.8
     - name: install tox
       run: python -m pip install tox
+    - uses: actions/checkout@v2
+    - name: pick to environment to run
+      run: |
+        python -c "print('::set-env name=TOXENV::' + {'3.8': 'py38', '3.7': 'py37', '3.6': 'py36', '3.5': 'py35', '2.7': 'py27', 'pypy2': 'pypy', 'pypy3': 'pypy3'}['${{ matrix.py}}'])"
```

Shocking but you don't really need this terrible nesting:
```suggestion
      run: |
        print(
            '::set-env name=TOXENV::' +
            {'3.8': 'py38', '3.7': 'py37', '3.6': 'py36', '3.5': 'py35', '2.7': 'py27', 'pypy2': 'pypy', 'pypy3': 'pypy3'}['${{ matrix.py}}']
        )
      shell: python
```

---

(I personally prefer putting this into the matrix tho)

---

By the way, why don't you just do static `TOXENV=python` and manage pythons on the job matrix level? Any problems with that?

---

Because very soon tox will not support running with old pythons, and I prefer tox running on latest python as opposed to whatever the test python needs to be (python 2.7 and such)